### PR TITLE
Fix TheAdmin & TheTheme for RTL languages

### LINKED ISSUE #14483: RTL Issues
After the last update in Bootstrap I noticed there were many things broken in RTL especially, did we test before merging it?

I'm not sure if CSS classes were added inline intentionally or for a purpose because all know the classes that affect the language direction is matter

/cc @agriffard @MikeAlhayek @Skrypt

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #14483 

@hishamco can you please test drive this PR with RTL language?

image

image

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
@hishamco what is the trick to turn the front end site into Arabic? I thought changing the Admin to Arabic should also change for the frontend but it did not. Not sure what is the trick. I do not use Arabic

**Comment by agriffard** [comment]
> @hishamco what is the trick to turn the front end site into Arabic? I thought changing the Admin to Arabic should also change for the frontend but it did not. Not sure what is the trick. I do not use Arabic

Setting `ar` as default culture in the admin works for me to display the front end site as RTL.
You can also call a front url and add `?culture=ar`

**Comment by agriffard** [comment]
Do not use the Admin culture picker that I can see on your screenshot, it only has an impact on the admin culture.

**Comment by hishamco** [comment]
@MikeAlhayek as @agriffard said no need to use the `AdminCulturePicker` what you need is a cookie or querystring to set the current culture

**Comment by MikeAlhayek** [comment]
@agriffard @hishamco I still cant switch the language for the front end. There has to be an easy way to switch language for the front end.

image

**Comment by hishamco** [comment]
Make sure you enable the localization module

**Comment by MikeAlhayek** [comment]
It's already enabled

image

image

**Comment by agriffard** [comment]
My admin menu and topbar does not react has expected on RTL, though:

image

**Comment by MikeAlhayek** [comment]
Make sure you use this PR. Also, make sure you do a deep refresh CTRL + F5 to see the changes I made in this PR.

**Comment by hishamco** [comment]
This might happen for one reason if the the default language in your browser is en-US, try to change it from OC to `it`and it should work. This is because HTTP Headers has a priority over querystring

**Comment by MikeAlhayek** [comment]
@hishamco I am not following you. Admin seems to work and switches to AR but the front end does not. I enabled `OrchardCore.ContentLocalization.ContentCulturePicker` feature, that did not add a dropdown menu on the front end as expected. Maybe there is a missing filter to inject the menu into the front end.

**Comment by hishamco** [comment]
> @hishamco I am not following you. Admin seems to work and switches to AR but the front end does not. I enabled `OrchardCore.ContentLocalization.ContentCulturePicker` feature, that did not add a dropdown menu on the front end as expected. Maybe there is a missing filter to inject the menu into the front end.
> 

Yep there's a shape you need to inject, I remember I showed everything in the last OC Harvest, you could check the docs too

**Comment by MikeAlhayek** [comment]
@hishamco I finally got the front end to display the menu and I changed the language. All seems to be working. I don't think asking the user to inject `ContentCulturePicker` in their theme is a very good idea. This shape could be injected using a filter in the `NavbarTop` zone.
Or better yet, create a new shape for the NavbarShape where we can use drivers to inject items into the menu. Similar to what I did in the UserMenu. Anyway, thats a subject for another day. I may submit a PR to improve this.

Do you still see issues with RTL?

**Comment by agriffard** [comment]
Dark theme has some light areas (Top, searches) on your branch:

image

**Comment by MikeAlhayek** [comment]
>  Or better yet, create a new shape for the NavbarShape where we can use drivers to inject items into the menu. Similar to what I did in the UserMenu. Anyway, thats a subject for another day. I may submit a PR to improve this.

I actually implemented this #14488 Much cleaner!

**Comment by MikeAlhayek** [comment]
@agriffard I fixed the issue we had with colors. Can you confirm that all issues have been fixed?

@hishamco please use the latest and let me know if you notice any issues.

**Comment by hishamco** [comment]
> @hishamco please use the latest and let me know if you notice any issues.

Sure

**Comment by MikeAlhayek** [comment]
@agriffard I think we should merge this PR to fix the all (or at least the major) issues with RTL.

We can always address minor RTL issues in separate PRs. thoughts?

**Comment by agriffard** [comment]
Please review https://docs.orchardcore.net/en/latest/docs/reference/modules/Localize/#shapes

image

You say that `AdminCulturePickerContainer` has been removed in the release notes 1.8 but you changed AdminCulturePickerContainer.cshtml instead of AdminCulturePicker.cshtml.

In the admin, it is still a select:

image

**Comment by MikeAlhayek** [comment]
@agriffard fixed that.

**Comment by hishamco** [comment]
@MikeAlhayek please give me some time to review this before you merge to avoid any consequences, moreover I will keep opening RTL issues as much as I can and prepare a proper fix

**Comment by hishamco** [comment]
> You say that AdminCulturePickerContainer has been removed in the release notes 1.8 but you changed AdminCulturePickerContainer.cshtml instead of AdminCulturePicker.cshtml.

Why it's removed?!!

**Comment by hishamco** [comment]
I might check this today

> Both AdminCulturePickerContainer and ContentCulturePickerContainer were removed by PR https://github.com/OrchardCMS/OrchardCore/pull/14488 for simplifications.

Can you show how this could be simplified?

**Review by agriffard — APPROVED** [review]
LGTM in order to already fix most RTL issues.

**Comment by MikeAlhayek** [comment]
> I might check this today
@hishamco sounds good. if we don't here back from you by Monday, will merge it then. We need to release this fix as it should fix most RTL issues (at least the visible issues)

> Can you show how this could be simplified?

You can either view the code changes or test drive it to see it in action. But, basically we now create the Navbar using driver instead of having to inject the shapes into `NavbarTop` zone or asking the user to modify themes just so they can see the content picker.

**Comment by hishamco** [comment]
It looks good but it seems there's an issue in the dropdown menu

File: src/docs/reference/modules/Localize/README.md
Comment (hishamco) [inline src/docs/reference/modules/Localize/README.md:172]: This is behaviour breaking change

File: src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs
Comment (hishamco) [inline src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs:20]: No need use

https://github.com/OrchardCMS/OrchardCore/blob/48b350b9a6c2986a3ffcf4b37c59a7bd3f58dc5c/src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/CultureInfoExtensions.cs#L22C24-L22C37

File: src/OrchardCore.Modules/OrchardCore.Localization/Views/AdminCulturePicker.cshtml
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Localization/Views/AdminCulturePicker.cshtml:4]: This might goes into another PR, the PR should fix the RTL issues, right?

File: src/OrchardCore.Themes/TheTheme/Views/Layout.cshtml
Comment (hishamco) [inline src/OrchardCore.Themes/TheTheme/Views/Layout.cshtml:25]:
```suggestion
    @if (CultureInfo.CurrentUICulture.IsRightToLeft())
```

File: src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml
Comment (hishamco) [inline src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml:29]:
```suggestion
    @if (CultureInfo.CurrentUICulture.IsRightToLeft())
```

File: src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml
Comment (hishamco) [inline src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml:31]: It could be simplified by ternary operator

**Review by hishamco — CHANGES_REQUESTED** [review]

File: src/docs/reference/modules/Localize/README.md
Comment (MikeAlhayek) [reply src/docs/reference/modules/Localize/README.md:172]: @hishamco how is this a breaking change?

File: src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs:20]: @hishamco 
This is as helpful as using the existing `CultureDir()`  or `CultureName()`. Using the `IsRightToLeft()` accessing the `IsRightToLeft()` method directly will require you to use to additional using statements in every view.

File: src/OrchardCore.Modules/OrchardCore.Localization/Views/AdminCulturePicker.cshtml
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Localization/Views/AdminCulturePicker.cshtml:4]: It could, but since @agriffard requested it, I am leaving it here. + Its an improvement to the theme itself which is what this PR is doing.

File: src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml
Comment (MikeAlhayek) [reply src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml:31]: ?

File: src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs
Comment (hishamco) [reply src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs:20]: You can use `CultureDir() == LanguageDirection.RTL` instead of introducing a new API that wraps another one

File: src/docs/reference/modules/Localize/README.md
Comment (hishamco) [reply src/docs/reference/modules/Localize/README.md:172]: `ContentCulturePicker` & `AdminCulturePicker` have been used for awhile, removing them will breaks an existing apps

File: src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml
Comment (hishamco) [reply src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml:31]: Just leave it

File: src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs:20]: I rather keep it simple and use the `IsRightToLeft()` on IOrchardHelper

File: src/docs/reference/modules/Localize/README.md
Comment (MikeAlhayek) [reply src/docs/reference/modules/Localize/README.md:172]: we did not remove `ContentCulturePicker` or `AdminCulturePicker`. In a different PR we removed `ContentCulturePickerContainer` and `AdminCulturePickerContainer` which did nothing more than creating a the actual shapes `ContentCulturePicker` and `AdminCulturePicker`. It would me much more helpful to actually test the changes than going back and fourth with the details.

File: src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs
Comment (hishamco) [reply src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs:20]: Let us see @sebastienros thought

**Comment by MikeAlhayek** [comment]
Merging this PR. Let's open up a new issue if any appears.

File: src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs
Comment (hishamco) [reply src/OrchardCore/OrchardCore.DisplayManagement/Extensions/RazorHelperExtensions.cs:20]: @MikeAlhayek I think you were ignoring this comment and merging the PR without waiting for others' feedback which is not good, especially in OSS

**Pull request merged**
